File: binder/c++_binder/test_client.cpp
#define LOG_TAG "HelloService"

//#define LOG_NDEBUG 0

#include <fcntl.h>
#include <sys/prctl.h>
#include <sys/wait.h>
#include <binder/IPCThreadState.h>
#include <binder/ProcessState.h>
#include <binder/IServiceManager.h>
#include <cutils/properties.h>
#include <utils/Log.h>
#include "IHelloService.h"


using namespace android;

int main(int argc __unused, char** argv)
{
	int ret  = -1;
	sp<ProcessState> proc(ProcessState::self());
	sp<IServiceManager> sm = defaultServiceManager();	
	sp<IBinder> binder = sm->getService(String16("hello"));
	if (binder == 0) {
		ALOGI("test_client getService failed");
		return -1;
	}
	sp<IHelloService> service =
	interface_cast<IHelloService>(binder);

	if(argc == 2) {
		service->sayhello();
		ALOGI("client call sayhello");
	}else if (argc == 3){
		int cnt = service->sayhello_to(argv[2]);
		ALOGI("client call sayhello_to, cnt = %d", cnt);
	}
	ALOGI("test_client: main");
    
    
}

File: binder/java_binder/TestClient.java

import android.util.Slog;
import android.os.IBinder;
import android.os.ServiceManager;


public class TestClient
{
	private static final String TAG = "TestClient";

	public static void main(String[] args)
	{
		System.out.println("Usage: need parameter: <hello|goodbye> [name]");
		Slog.i(TAG,"get hello server");
		IBinder binder = ServiceManager.getService("hello");
		if (binder == null)
        {
            System.out.println("can not get hello service");
            Slog.i(TAG, "can not get hello service");
            return;
        }
		IHelloService service = IHelloService.Stub.asInterface(binder);

		if(args.length == 1){
			Slog.i(TAG,"client sayhello");
			try{
				service.sayhello();
			}catch(Exception e){}
		}
		else if(args.length == 2){
			Slog.i(TAG,"client sayhello_to");
			try{
				service.sayhello_to(args[1]);
			}catch(Exception e){}
		}
		
	}

}









